instantsearch/src/main/java/com/algolia/instantsearch/helpers/Searcher.java

```diff
@@ -103,55 +106,116 @@
+    }
+
+    /**
+     * Constructs a Searcher, creating its {@link Searcher#index} and {@link Searcher#client} with the given parameters.
+     *
+     * @param appId     your Algolia Application ID.
+     * @param apiKey    a search-only API Key. (never use API keys that could modify your records! see https://www.algolia.com/doc/guides/security/api-keys)
+     * @param indexName the name of the index to target.
+     * @param variant   an identifier to differentiate this Searcher from eventual others.
+     * @return the new instance.
+     */
+    @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
+    public static Searcher create(@NonNull final String appId, @NonNull final String apiKey, @NonNull final String indexName, @NonNull String variant) {
+        return create(new Client(appId, apiKey).getIndex(indexName), variant);
     }
 
     /**
-     * Constructs the Searcher from an existing {@link Index}.
+     * Constructs a Searcher from an existing {@link Index}.
      *
      * @param index an Index initialized and eventually configured.
      * @return the new instance.
      */
     @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
     public static Searcher create(@NonNull final Index index) {
+        final String key = index.getIndexName();
+        Searcher instance = instances.get(key);
         if (instance == null) {
-            instance = new Searcher(index);
+            instance = new Searcher(index, key);
+            instances.put(key, instance);
+        } else {
+            throw new IllegalStateException("There is already a Searcher for index " + key + ", you must specify a variant.");
         }
         return instance;
     }
 
     /**
-     * Constructs an helper, creating its {@link Searcher#index} and {@link Searcher#client} with the given parameters.
+     * Constructs the Searcher from an existing {@link Index}, eventually replacing the existing searcher for {@code variant}.
      *
-     * @param appId     Your Algolia Application ID.
-     * @param apiKey    A search-only API Key. (never use API keys that could modify your records! see https://www.algolia.com/doc/guides/security/api-keys)
-     * @param indexName An index to target.
+     * @param variant an identifier to differentiate this Searcher from eventual others using the same index. Defaults to the index's name.
+     * @param index   an Index initialized and eventually configured.
      * @return the new instance.
      */
     @SuppressWarnings({"WeakerAccess", "unused"}) // For library users
-    public static Searcher create(@NonNull final String appId, @NonNull final String apiKey, @NonNull final String indexName) {
-        return create(new Client(appId, apiKey).getIndex(indexName));
+    public static Searcher create(@NonNull final Index index, @NonNull String variant) {
+        boolean shouldReplace = false;
+        Searcher instance = instances.get(variant);
+        if (instance != null) {
```

Comment:
Instead of 2 ifs:

if (instance == null || instance.getIndex() != index) { .. replace logic }

Less code and no temp variable